File: EpillBox.API/Models/FirstAidKitMedicine.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace EpillBox.API.Models
{
    public class FirstAidKitMedicine
    {
        public int FirstAidKitMedicineID { get; set; }

        [ForeignKey("Medicine")]
        public int MedicineID { get; set; }

        [ForeignKey("FirstAidKit")]
        public int FirstAidKitID { get; set; }

        public Medicine Medicine { get; set; }
        public FirstAidKit FirstAidKit { get; set; }

        public DateTime? ExpirationDate { get; set; }
        public bool IsTaken { get; set; }
        public int RemainingQuantity { get; set; }
       
        public DateTime FirstServingAt {get; set;}
        public int NumberOfServings { get; set; }
        public int ServingSize { get; set; }
       
    }
}

File: EpillBox-SPA/src/app/_model/FirstAidKitMedicine.ts
export interface FirstAidKitMedicine {
  expirationDate?: Date | null;
  firstAidKitID?: number | null;
  firstAidKitMedicineID?: number | null;
  isTaken?: boolean | null;
  name: string;
  medicineID: number;
  remainingQuantity?: number | null;
  quantityInPackage?: number | null;
  fakName?: string | null;
  form?: string | null;
  producer?: string | null;
  activeSubstance?: string[] | null;
}
